Improve the output of the swagger API for watch events

smarterclayton opened the pull request:
Stopgap to improve this prior to converting watch resources to
versioned objects.

Spawned from #6182

bgrant0607 commented:
LGTM. Thanks. @thockin can rebase on top of this and regenerate the swagger.

brendandburns commented:
Tests are failing:
```
# github.com/GoogleCloudPlatform/kubernetes/pkg/watch/json
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/watch/json/decoder_test.go:45: undefined: watchEvent
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/watch/json [build failed]
```

bgrant0607 commented:
Merged multi-port services. Please run hack/update-swagger-spec.sh.

smarterclayton commented:
Rebased, test fixed, and docs updated.

----- Original Message -----
> Merged multi-port services. Please run hack/update-swagger-spec.sh.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6232#issuecomment-88226844

bgrant0607 commented on api/swagger-spec/v1beta3.json:
```diff
@@ -7276,38 +7483,10 @@
     "id": "v1beta3.ServiceSpec",
     "required": [
-     "ports",
+     "port",
```
Something's not right here. This is reverting multi-port services, among other things.

smarterclayton replied:
Bad merge, let me delete and regen.

----- Original Message -----
> >     "v1beta3.ServiceSpec": {
> >      "id": "v1beta3.ServiceSpec",
> >      "required": [
> > -     "ports",
> > +     "port",
> 
> Something's not right here. This is reverting multi-port services, among
> other things.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6232/files#r27528460

smarterclayton replied:
Fixed, sorry.

----- Original Message -----
> Bad merge, let me delete and regen.
> 
> ----- Original Message -----
> > >     "v1beta3.ServiceSpec": {
> > >      "id": "v1beta3.ServiceSpec",
> > >      "required": [
> > > -     "ports",
> > > +     "port",
> > 
> > Something's not right here. This is reverting multi-port services, among
> > other things.
> > 
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/GoogleCloudPlatform/kubernetes/pull/6232/files#r27528460
>

bgrant0607 commented:
Weird. I'm not seeing the comments I received via email, nor the latest push.

bgrant0607 commented:
@smarterclayton Maybe try to make clean, rebuild, and regenerate the swagger?

@nikhiljindal Any guess about what's going on with the swagger generation? A stale go-restful dependency?

nikhiljindal commented:
@smarterclayton : Did you rebuild the binary before running update-swagger-spec?
That script uses the existing binary rather then re-building it.

bgrant0607 commented:
@nikhiljindal He picked up some recent API changes, so I'd guess he did build it.

nikhiljindal commented:
The type changes and addition of a path parameter in swagger spec are coming from #6143 and #6117.

bgrant0607 commented:
Ok, sorry. Swagger has been updated from earlier changes. Please rebase and re-update swagger again. Feel free to self-merge.

bgrant0607 commented:
@smarterclayton Could we try to get this in before something breaks it again? :-) It might be easier if you didn't update the swagger in this PR.

smarterclayton committed:
Improve the output of the swagger API for watch events

Stopgap to improve this prior to converting watch resources to
versioned objects.

smarterclayton committed:
Update swagger spec

smarterclayton commented:
Rebased, split swagger out.

bgrant0607 commented:
LGTM. Thanks.

When looking at this I noticed that the /log content types are wrong.

The shippable failure is a known kubelet test problem.
